src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -427,12 +427,18 @@
     <Field Name="GetResult" Type="MethodSymbol" Null="allow"/>
   </Node>
 
-  <Node Name="BoundTypeOfOperator" Base="BoundExpression">
+  <AbstractNode Name="BoundTypeOf" Base="BoundExpression">
     <!-- Non-null type is required for this node kind -->
     <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
-    <Field Name="SourceType" Type="BoundTypeExpression"/>
     <!-- Well-known member populated during lowering -->
     <Field Name="GetTypeFromHandle" Type="MethodSymbol" Null="allow"/>
+  </AbstractNode>
+    
+  <Node Name="BoundTypeOfOperator" Base="BoundTypeOf">
+    <Field Name="SourceType" Type="BoundTypeExpression"/>
+  </Node>
+    
+  <Node Name="BoundTypeOfPrivateImplementationDetails" Base="BoundTypeOf">
```

Comment:
This node kind is used to synthesize `typeof(PrivateImplementationDetails)` in the static constructor of `PrivateImplementationDetails` without referring to the type?